packages/babel-preset/src/transformPlugin.ts

```diff
@@ -62,6 +77,90 @@ function getCalleeFunctionKind(
   return null;
 }
 
+/**
+ * Gets the id of the parent variable declarator
+ */
+function getParentDeclaratorId(path: NodePath<t.CallExpression> | NodePath<t.MemberExpression>): string {
+  const declarator = path.findParent(p => p.isVariableDeclarator());
+  if (declarator?.isVariableDeclarator()) {
+    const id = declarator.get('id');
+    if (id.isIdentifier()) {
+      return id.node.name;
+    }
+  }
+
+  return 'unknownHook';
+}
+
+/**
+ * Extracts CSS rules from evaluated reset styles to metadata
+ */
+function buildCSSResetEntriesMetadata(
+  state: BabelPluginState,
+  cssRules: string[] | CSSRulesByBucket,
+  hookName: string,
+) {
+  state.cssResetEntries ??= {};
+  state.cssResetEntries[hookName] ??= [];
+  if (Array.isArray(cssRules)) {
+    state.cssResetEntries[hookName] = cssRules;
+  } else {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, bucketEntries] of Object.entries(cssRules)) {
+      for (const bucketEntry of bucketEntries) {
+        if (Array.isArray(bucketEntry)) {
+          state.cssResetEntries[hookName].push(bucketEntry[0]);
+        } else {
+          state.cssResetEntries[hookName].push(bucketEntry);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Extracts CSS rules from evaluated styles to metadata
+ */
+function buildCSSEntriesMetadata(
+  state: BabelPluginState,
+  classnamesMapping: CSSClassesMapBySlot<string>,
+  cssRulesByBucket: CSSRulesByBucket,
+  hookName: string,
+) {
+  const classesBySlot: Record<string, string[]> = {};
+  for (const [slot, cssClassesMap] of Object.entries(classnamesMapping)) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, cssClasses] of Object.entries(cssClassesMap)) {
+      classesBySlot[slot] = Array.isArray(cssClasses) ? cssClasses : [cssClasses];
+    }
+  }
```

Comment:
This part is broken: 

<img width="1558" alt="image" src="https://github.com/microsoft/griffel/assets/14183168/53cca335-c528-4ad0-8362-14d9eef63f77">